src/App.jsx

```diff
@@ -40,18 +41,44 @@ function App() {
         }
       }
     }, {})
-  }, [responseArray])
+  }, [reviveResponseArray])
 
-  const reviveData = useMemo(() => {
-    return Object.values(usersRevivedAsObject)
-  }, [usersRevivedAsObject])
+  const parsedUserEventsAsObject = useMemo(() => {
+    const getIdRegex = /XID\=([0-9]+)(?:"\>|\>)(.+)\<.*/gm;
+    const stripAnchorRegex = /(<\/?[a|A][^>]*>|)/gm;
+    return eventResponseArray
+      .map(item => ({
+        match: getIdRegex.exec(item),
+        message: item.replaceAll(stripAnchorRegex, "")
+      }))
+      .filter(item => item.match !== null)
+      .map(item => {
+        return {
+          id: parseInt(item.match[1]),
+          message: item.message
+        }
+      })
+  }, [eventResponseArray])
+
+  const finalData = useMemo(() => {
+    return Object.values(usersRevivedAsObject).map(item => {
```

Comment:
MISC: You could have `Object.values(usersRevivedAsObject)` and `Object.values(parsedUserEventsAsObject)` as memos that return arrays so you don't need to do it in here.

It's much of a muchness and really up to your preference.